### PR TITLE
[ITBL-3976] Slight refactor of action buttons payload

### PR DESCRIPTION
**Pull request by vbabenkoru**
Update action buttons payload: specify whether to open the app via 'openApp' and have buttonType only control the appearance of the button

### CODE REVIEW

File: IterableAppExtensionsTests/IterableAppExtensionsTests.m [hunk 1]
```diff
@@ -252,6 +257,49 @@ - (void)testPushTextInputButton {
...
+    XCTestExpectation *expectation = [[XCTestExpectation alloc] initWithDescription:@"contentHandler is called"];
+    
+    [self.extension didReceiveNotificationRequest:request withContentHandler:^(UNNotificationContent *contentToDeliver) {
+        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, IterableNotificationCenterRequestDelay * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
+            UNUserNotificationCenter* center = [UNUserNotificationCenter currentNotificationCenter];
+            [center getNotificationCategoriesWithCompletionHandler:^(NSSet<UNNotificationCategory *> * _Nonnull categories) {
+                UNNotificationCategory *createdCategory = nil;
+                for (UNNotificationCategory *category in categories) {
+                    if ([category.identifier isEqualToString:content.userInfo[@"itbl"][@"messageId"]])
+                        createdCategory = category;
```
Comment (tapashmajumder) [inline IterableAppExtensionsTests/IterableAppExtensionsTests.m]: Just a suggestion and not necessary for PR approval. I like open/close brackets after single line if statements even though it is not necessary.

Comment (vbabenkoru) [reply IterableAppExtensionsTests/IterableAppExtensionsTests.m]: Right! I'm so used to it I don't notice.. I'll see if there are any style checks in AppCode for that.

Comment (davidtruong) [reply IterableAppExtensionsTests/IterableAppExtensionsTests.m]: Same I like the literal brackets for clarity

File: IterableAppExtensionsTests/IterableAppExtensionsTests.m [hunk 2]
```diff
@@ -176,8 +180,8 @@ - (void)testPushDestructiveSilentActionButton {
                                  @"actionButtons": @[@{
                                                          @"identifier": @"destructiveButton",
                                                          @"title": @"Unsubscribe",
-                                                         @"destructive": @YES,
-                                                         @"buttonType": @"dismiss",
+                                                         @"buttonType": @"destructive",
+                                                         @"openApp": @NO,
```
Comment (davidtruong) [inline IterableAppExtensionsTests/IterableAppExtensionsTests.m:185]: Will this still work if we don't specify an openApp or the action in the payload?

**Review by davidtruong — APPROVED** [review]
LFTM

File: IterableAppExtensionsTests/IterableAppExtensionsTests.m (hunk 2 above)
Comment (vbabenkoru) [reply IterableAppExtensionsTests/IterableAppExtensionsTests.m:185]: Yup, ITBNotificationServiceExtension.m line 129.

File: IterableAppExtensions/ITBNotificationServiceExtension.m [hunk 3]
```diff
@@ -122,22 +122,26 @@ - (UNNotificationAction *)createNotificationActionButton:(NSDictionary *)buttonD
     NSString *identifier = buttonDictionary[ITBL_BUTTON_IDENTIFIER];
     NSString *title = buttonDictionary[ITBL_BUTTON_TITLE];
     NSString *buttonType = buttonDictionary[ITBL_BUTTON_TYPE];
-    if (!buttonType) {
-        buttonType = IterableButtonTypeOpen;
+    if (!buttonType || (![buttonType isEqualToString:IterableButtonTypeTextInput] &&
+                        ![buttonType isEqualToString:IterableButtonTypeDestructive])) {
+        buttonType = IterableButtonTypeDefault;
+    }
+    BOOL openApp = YES;
```
Comment (vbabenkoru) [inline IterableAppExtensions/ITBNotificationServiceExtension.m:129]: Default value for openApp

**Pull request merged**